inc/classes/subscriber/preload/class-preload-subscriber.php

```diff
@@ -0,0 +1,217 @@
+<?php
+namespace WP_Rocket\Subscriber\Preload;
+
+use WP_Rocket\Event_Management\Subscriber_Interface;
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Preload\Homepage;
+
+/**
+ * Preload Subscriber
+ *
+ * @since 3.2
+ * @author Remy Perona
+ */
+class Preload_Subscriber implements Subscriber_Interface {
+	/**
+	 * Homepage Preload instance
+	 *
+	 * @since 3.2
+	 * @author Remy Perona
+	 *
+	 * @var Homepage
+	 */
+	private $homepage_preloader;
+
+	/**
+	 * WP Rocket Options instance.
+	 *
+	 * @since 3.2
+	 * @author Remy Perona
+	 *
+	 * @var Options_Data
+	 */
+	private $options;
+
+	/**
+	 * Constructor
+	 *
+	 * @since 3.2
+	 * @author Remy Perona
+	 *
+	 * @param Homepage     $homepage_preloader Homepage Preload instance.
+	 * @param Options_Data $options            WP Rocket Options instance.
+	 */
+	public function __construct( Homepage $homepage_preloader, Options_Data $options ) {
+		$this->homepage_preloader = $homepage_preloader;
+		$this->options            = $options;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'admin_notices'                   => [
+				[ 'notice_preload_running' ],
+				[ 'notice_preload_complete' ],
+			],
+			'admin_post_rocket_stop_preload'  => [ 'do_admin_post_stop_preload' ],
+			'rocket_purge_time_event'         => [ 'run_preload', 11 ],
+			'pagely_cache_purge_after'        => [ 'run_preload', 11 ],
+			'update_option_' . WP_ROCKET_SLUG => [
+				[ 'maybe_launch_preload', 11, 2 ],
+				[ 'maybe_cancel_preload', 10, 2 ],
+			],
+		];
+	}
+
+	/**
+	 * Launches the homepage preload
+	 *
+	 * @since 3.2
+	 * @author Remy Perona
+	 *
+	 * @param string $lang The language code to preload.
+	 * @return void
+	 */
+	protected function preload( $lang = '' ) {
+		$urls = get_rocket_i18n_uri();
+
+		if ( $lang ) {
+			$urls = (array) get_rocket_i18n_home_url( $lang );
```

Comment:
Would be more efficient to do this I think:
```php
if ( $lang ) {
    $urls = (array) get_rocket_i18n_home_url( $lang );
} else {
    $urls = get_rocket_i18n_uri();
}
```
`run_rocket_bot()` does a similar thing.